.github/workflows/ci.yml

```
@@ -0,0 +1,25 @@
+name: CI
+
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+
+jobs:
+    build:
+
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                node-version: [14.x, 16.x, 18.x, 20.x]
+        
+        steps:
+            - uses: actions/checkout@v3
+            - name: Use Node.js ${{ matrix.node-version }}
+              uses: actions/setup-node@v3
+              with:
+                node-version: ${{ matrix.node-version }}
+            - run: npm ci
+            - run: npm run build --if-present
+            - run: npm test
```

```suggestion
    build:
        runs-on: ubuntu-latest        
        steps:
            - uses: actions/checkout@v3
            - name: Use Node.js
              uses: actions/setup-node@v3
              with:
                node-version: 20
            - run: npm ci
            - run: npm run build
```

```yml
strategy:
    matrix:
        node-version: [14.x, 16.x, 18.x, 20.x]
```

のところは、Node.js のいろいろなバージョンでテストするものなので、今回ならなくても良いかと。

```yml
with:
  node-version: ${{ matrix.node-version }}
```

もそれに合わせて、`20` とか `latest` とか適当なバージョンを設定すればうまくいきそう。

```yml
- run: npm run build --if-present
- run: npm test
```

ここは、何をしたいかだけど、ビルドをしたいだけだったら、`- run: npm run build` で十分かな。`npm test` は定義されてないから、エラーを吐いてる。

---

コピペしただけだから、何やってるかよく分かってないのです。

---

www

---

これを参考にしてください。
https://github.com/ut-code/learn-braille/blob/main/.github/workflows/ci.yml

---

```yml
build:
    name: Build
    runs-on: ubuntu-latest
    steps:
      - name: Checkout Code
        uses: actions/checkout@v3

      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: latest
          cache: npm

      - name: Install Node Packages
        run: npm ci
        shell: bash

      - name: Build
        run: npm run build
        shell: bash
```
ここのところを無限に書き足せば、いろんな CI が作れるよって感じの認識でいいかと。